src/linear_algebra/dual.lean

```diff
@@ -558,3 +558,81 @@ end
 end
 
 end subspace
+
+namespace linear_map
+
+variables {R : Type*} [comm_ring R] {M₁ : Type*} {M₂ : Type*}
+variables [add_comm_group M₁] [module R M₁] [add_comm_group M₂] [module R M₂]
+
+open module
+
+/-- Given a linear map `f : M₁ →ₗ[R] M₂`, `f.dual_map` is the linear map between the dual of
+`M₂` and `M₁` such that it maps the functional `φ` to `φ ∘ f`. -/
+def dual_map (f : M₁ →ₗ[R] M₂) : dual R M₂ →ₗ[R] dual R M₁ :=
+lcomp R R f
```

Comment:
Can you add a version for `linear_equiv` too?
```lean
def linear_equiv.dual_map (f : M₁ ≃ₗ[R] M₂) : dual R M₂ ≃ₗ[R] dual R M₁ :=
{ to_fun := dual_map f, inv_fun := dual_map f.symm, sorry }
```

Reply:
done!